lib/transaction/transaction.go

```diff
@@ -61,15 +61,47 @@ func (t *Transaction) UnmarshalJSON(b []byte) (err error) {
 	return
 }
 
+func (tx Transaction) AdjustFeeWithFrozenAccount(isSourceLinked bool) (err error) {
+	fee := common.FrozenFee
+	if !isSourceLinked {
+		var opsHaveFee int
+		for _, op := range tx.B.Operations {
+			if op.HasFee(isSourceLinked) {
+				opsHaveFee++
+			}
+		}
+		if opsHaveFee < 1 {
+			return
+		}
+		fee = common.BaseFee.MustMult(opsHaveFee)
+	}
+	tx.B.Fee = fee
+	tx.H.Hash = tx.B.MakeHashString()
+	return
+}
+
+func NewTransactionAdujustFeeWithFrozenAccount(source string, sequenceID uint64, isSourceLinked bool, ops ...operation.Operation) (tx Transaction, err error) {
+	if tx, err = NewTransaction(source, sequenceID, ops...); err != nil {
+		return
+	}
+	tx.AdjustFeeWithFrozenAccount(isSourceLinked)
```

Comment:
What I wanted is like this
```
tx = tx.AdjustFeeWithFrozenAccount(isSourceLinked)
```

If you want to do like what you've done, then please change
```
func (tx Transaction) AdjustFeeWithFrozenAccount(isSourceLinked bool) (err error)
```
to
```
func (tx *Transaction) AdjustFeeWithFrozenAccount(isSourceLinked bool) (err error)
```

Reply:
Ok I understand it.